src/components/LiquidityDepth/DensityChart.tsx

```diff
@@ -0,0 +1,191 @@
+
+  interactive = interactive && Boolean(formattedData?.length)
+
+  const filteredData =
+    activeChartEntry && zoom
+      ? formattedData?.slice(Math.max(0, activeChartEntry.index - zoom), activeChartEntry.index + zoom)
+      : undefined
+
+  return (
+    <Wrapper>
+      {/* filteredData === undefined will show sample data */}
+      {filteredData === [] ? (
+        <ColumnCenter>
+          <XCircle stroke={theme.text4} />
+          <TYPE.darkGray padding={10}>
+            <Trans>No data</Trans>
+          </TYPE.darkGray>
+        </ColumnCenter>
+      ) : (
+        <VictoryChart
+          height={275}
+          padding={40}
+          minDomain={{ y: 0 }}
+          containerComponent={
+            <VictoryBrushContainer
+              allowDraw={false}
+              allowDrag={interactive}
+              allowResize={interactive}
+              brushDimension="x"
+              brushDomain={
+                leftPrice && rightPrice
+                  ? {
+                      x: [parseFloat(leftPrice?.toSignificant(5)), parseFloat(rightPrice?.toSignificant(5))],
+                    }
+                  : undefined
+              }
+              brushComponent={
+                <Brush
+                  leftHandleColor={currencyA ? tokenAColor : theme.primary1}
+                  leftLabel={
+                    price && leftPrice
+                      ? `${((parseFloat(leftPrice.toSignificant(8)) / parseFloat(price) - 1) * 100).toFixed(2)}%`
+                      : undefined
+                  }
+                  rightLabel={
+                    price && rightPrice
+                      ? `${((parseFloat(rightPrice.toSignificant(8)) / parseFloat(price) - 1) * 100).toFixed(2)}%`
+                      : undefined
+                  }
+                  rightHandleColor={currencyB ? tokenBColor : theme.secondary1}
+                  allowDrag={interactive}
+                />
+              }
+              onBrushDomainChangeEnd={(domain) => {
+                const leftRangeValue = Number(domain.x[0])
+                const rightRangeValue = Number(domain.x[1])
+
+                // simulate user input for auto-formatting and other validations
+                leftRangeValue > 0 && onLeftRangeInput(leftRangeValue.toFixed(6))
+                rightRangeValue > 0 && onRightRangeInput(rightRangeValue.toFixed(6))
+              }}
+              handleWidth={30 /* handle width must be as large as handle head */}
+            />
+          }
+        >
+          <VictoryArea
+            data={filteredData ? filteredData : sampleData}
+            style={{ data: { stroke: theme.blue1, fill: theme.blue1, opacity: '0.5' } }}
+            x={'price0'}
+            y={'activeLiquidity'}
+            interpolation="step"
+          />
+
+          {price && (
+            <VictoryLine
+              data={
+                maxLiquidity && price
+                  ? [
+                      { x: parseFloat(price), y: 0 },
```

Comment:
i think this only works because it tries to use `toString` on price? ideally should always specify e.g. `toSignificant`

Reply:
price is actually a string here.. I think I agree that I shouldn't pass as a string but pass the Price<> and let the component format.